### DISCUSSION TITLE
disable submit button if required filleds are empty

### DISCUSSION THREAD

**Discussion by IlitA** [react-hook-form/react-hook-form#1714] [General] [answered]
Is it possible to enable the submit button only after all required fields have value without trigger validation on change? 

I'd like to render errors only on submit but disable the submit button if some of the inputs are empty.

**Comment by bluebill1049** [comment, marked as answer]
perhaps you will need to use `dirtyFields` to track all the inputs.

**Reply by IlitA** [reply]
@bluebill1049 thank you for the answer, but it won't work as not all fields are required.
I feel that checking if all required fields are in `dirtyFields` will be overkill.

**Reply by bluebill1049** [reply]
alternative uses `onChange` mode and shows errors with combination of `formState`. so basically validation kicks in anyhow and you control when to show error message.

**Reply by IlitA** [reply]
Maybe I can use the submit count. I will try. thanks

**Reply by bluebill1049** [reply]
worth to mention there is `isSubmitted` as well.

**Reply by IlitA** [reply]
it works! 🥳
